src/mlx/warnings.py

```diff
@@ -244,28 +221,20 @@ class JUnitChecker(WarningsChecker):
     def __init__(self):
         super(JUnitChecker, self).__init__(name=JUnitChecker.name)
 
-    def check_file(self, filename):
+    def check(self, content):
         '''
-        Function for counting the number of JUnit failures in a specific file
+        Function for counting the number of JUnit failures in a specific text
 
         Args:
-            filename (str): The path to the file to parse
+            content (str): The content to parse
         '''
-        with open(filename, 'r') as xmlfile:
-            ts, tr = xunitparser.parse(xmlfile)
+        try:
+            result = JUnitXml.fromstring(content)
+            result.update_statistics()
+            self.count += result.errors + result.failures
+        except Exception as e:
+            print('Exception {exception} while parsing JUnit'.format(exception=type(e).__name__))
```

Comment:
This will prevent the exception from propagating further and killing the program. Is that the intention?

If not, it might be useful to reraise the exception with the changed text? Like here: https://stackoverflow.com/a/9157277